### DISCUSSION TITLE
Inverted /dark mode support

### DISCUSSION THREAD

**Discussion by abhiz710** [zxing-cpp/zxing-cpp#424] [Q&A] [answered]
Hi, zxing supports inverted/dark mode in C# version. 
But in cpp version i see its not yet released. 
When can we expect it to be released?
@axxel ?

**Comment by abhiz710** [comment]
@axxel plz let me know

**Reply by axxel** [reply]
Sorry for letting you wait for reply. The thing is, that I currently don't know, yet. It is still in a bit of a hold up to make sure the ABI will be stable for a while after 2.0 is released. See #333 .

**Comment by axxel** [comment, marked as answer]
Version 2.0 with `DecodeHints::tryInvert()` support has been [released](https://github.com/zxing-cpp/zxing-cpp/releases/tag/v2.0.0).

**Reply by kientux** [reply]
Does this apply for linear codes? I try it with linear codes and it's not working. Currently I have to invert the color of input myself and read the second time, my app doesn't care too much about performance but if zxing-cpp can handle it then it would be way more convenience.

**Reply by axxel** [reply]
> Does this apply for linear codes?

No, it does not, because linear barcode symbologies do not support "reflectance reversal". See the discussion [here](https://github.com/zxing-cpp/zxing-cpp/issues/172#issuecomment-1228483623).
